src/main/kotlin/lotto/number/Numbers.kt

```diff
@@ -1,5 +1,15 @@
 package lotto.number
 
 data class Numbers(
-    val numbers: List<Int>,
-)
+    val numbers: List<LottoNumber>,
+) {
+    fun hasNumber(number: LottoNumber): Boolean = numbers.contains(number)
+
+    fun countMatching(other: Numbers): Int = other.numbers.count { this.hasNumber(it) }
+
+    companion object {
+        fun List<LottoNumber>.sorted(): List<LottoNumber> = this.sortedBy { it.number }
+
+        fun fromInts(ints: List<Int>): Numbers = Numbers(ints.map { LottoNumber(it) })
```

Comment:
이러한 로직은 부생성자로 만들었을 때 주생성자와 List 타입이 충돌되어 만들 수 없는데요, 혹시 이 사실을 인지하고 팩토리 함수를 만들어주신걸까요~?

아래와 같은 시각도 있더라고요!
https://kt.academy/article/ek-factory-functions

Reply:
네네! 처음에 부생성자로 만들었다가 타입 충돌로 컴파일 에러가 발생해서 팩토리 메서드로 작성했습니다!

Reply:
참고로 이러한 현상은 [Java type erasure](https://stackoverflow.com/a/339708)와 관련이 있는데요, [Effective Kotlin](https://kt.academy/article/ek-factory-functions)에서도 설명하는 것처럼 아래와 같이 operator overloading을 통해 생성자를 부르는 것처럼 만들 수는 있습니다.

```kotlin
class Numbers(
    val numbers: List<LottoNumber>,
) {
    // ...
    companion object {
         operator fun invoke(ints: List<Int>) = Numbers(ints.map { LottoNumber(it) })
    }
}
```

좋은 방법인지에 대해서는 의견이 분분합니다 😓 저는 개인적으로 선호하진 않아요!

> However, implementing invoke in a companion object to make a fake constructor is considered less idiomatic. I do not recommend it, primarily because it violates Item 11: An operator’s meaning should be consistent with its function name.